### PR TITLE
Jira dummy data

### LINKED ISSUE #2100: Source Jira: create dummy data for issue-related and project-related streams
## Tell us about the new integration you’d like to have
Populate each of these streams with 50+ objects

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368213940671) by [Unito](https://www.unito.io)

### LINKED ISSUE #2101: Source Jira: create dummy data for non issue-related or project-related streams
## Tell us about the new integration you’d like to have
Populate each of these streams with 50+ objects

### PR DESCRIPTION
**Pull request by vitaliizazmic**
## What
Populating streams with data

Closes #2100 and #2101

## Pre-merge Checklist
* [ ] _Run integration tests_
* [ ] _Publish Docker images_

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1200367912513076/1200368101667530) by [Unito](https://www.unito.io)

### CODE REVIEW

**Review by sherifnada — APPROVED** [review]

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    def list(self):
+        pass
+
+    def generate_dashboards(self, url):
+        """https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-dashboards/#api-rest-api-3-dashboard-post"""
+        for index in range(1, 20):
+            payload = json.dumps(
+                {
+                    "name": f"Test dashboard {index}",
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py:45]: I'm not sure, but I think this method will work only once. On restarting, we will receive errors from Jira API that a Dashboard with the same name already exists. Is this normal behavior going to be?

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/groups.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+    def extract(self, response):
+        pass
+
+    def list(self):
+        pass
+
+    def generate_groups(self, url):
+        for index in range(20):
+            payload = json.dumps({"name": f"Test group {index}"})
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/groups.py:42]: The same quastion

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issues.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    def list(self):
+        """https://developer.atlassian.com/cloud/jira/platform/rest/v3/api-group-issue-search/#api-rest-api-3-search-get"""
+        params = {}
+        url = self.get_url(endpoint=self.list_endpoint)
+        for issue in self.fetch_data(url, params):
+            yield issue
+
+    def generate(self):
+        pass
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issues.py:47]: Why aren't we generating the Issue objects?

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/project_components.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    def list(self):
+        pass
+
+    def generate_project_categories(self, url):
+        for project in self.get_projects():
+            for index in range(random.randrange(6)):
+                payload = json.dumps(
+                    {
+                        "isAssigneeTypeValid": False,
+                        "name": f"Component {index}",
```
Comment (yevhenii-ldv) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/project_components.py:48]: Same question as for Dashboard

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+
+from ..base import BaseStream
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py:27]: no relative imports

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py (hunk 1 above)
Comment (keu) [reply airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py:45]: agree here, I would invest more time in using faker here

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+
+class Dashboards(BaseStream):
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py:30]: also, I don't see why it is necessary to use Stream abstraction here to generate data

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+    list_endpoint = "dashboard"
+    generate_endpoint = "dashboard"
+
+    def extract(self, response):
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/dashboards.py:34]: this supports my prev comment

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issue_links.py [hunk 8]
```diff
@@ -0,0 +1,74 @@
...
+        for item in issues_batch:
+            project_key = item.get("fields").get("project").get("key")
+            outward_issue = random.randrange(2)
+            if outward_issue == 1:
+                outward_issues[project_key] = item
+            else:
+                if project_key not in outward_issues:
+                    outward_issues[project_key] = item
+                else:
+                    for index in range(1, random.randrange(2, 11)):
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issue_links.py:54]: +1 for using `faker` here

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issue_remote_links.py [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+import json
+
+from ..base import BaseStream
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams/issue_remote_links.py:27]: relative imports

**Review by keu — CHANGES_REQUESTED** [review]
In general good job, but I have concerns about the maintainability of this code.
Lets see if we can use `faker` here. and `click` or just `argparser`

**Comment by sherifnada** [comment]
@eugene-kulak @yevhenii-ldv I haven't read every comment so don't take this too literally, but the maintainability of the fixtures code is not _that_ important. The most important thing is to clearly document via code which API calls we ran to generate which data. This should not be expected to be a perfectly runnable script because we will probably run it maybe once or twice more. It's ok if this errors out. But the most valuable thing to deliver here is the connector itself, we shouldn't spend too much time on this script. Better to spend that time on the connector itself.

File: airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams.py [hunk 10]
```diff
@@ -0,0 +1,435 @@
...
+        for index in range(30):
+            payload = json.dumps(
+                {
+                    "defaultWorkflow": "jira",
+                    "name": f"Test workflow scheme {index}",
+                    "description": "The description of the example workflow scheme.",
+                }
+            )
+            self.generate_record(payload)
+
```
Comment (keu) [inline airbyte-integrations/connectors/source-jira/integration_tests/fixtures/data_generator/streams.py:430]: format

**Review by keu — APPROVED** [review]
lgtm, but run format please

**Pull request merged**